Changelog.md

```diff
@@ -1,10 +1,35 @@
 # Motoko compiler changelog
 
-* Improve recursive deserialization capacity to match recursive serialization capacity by reducing
-  Wasm stack consumption (#3809).
-  Because of the bounds on recursion depth imposed by fixed-size stack, the
-  advice remains the same: avoid deeply nested recursive data structures.
-  Think "shallow trees good, very long lists bad".
+
+* motoko (`moc`)
+
+  * new 'let-else' construct for handling pattern-match failure.
+
+    The let declaration `let <pat> = <exp> else <block-or-exp>` has type `T` and declares the bindings in `<pat>` provided:
+    -   `<exp>` has type `T`,
+    -   `<pat>` has type `T`, and
+    -   `<block-or-exp>` has type `None`.
+
+    The declaration `let <pat> = <exp> else <block-or-exp>` evaluates `<exp>` to a result `r`.
```

Comment:
Isn't this too detailed for a changelog entry? (It would be adequate for the manual.) Something like "New feature `let-else` allows to take a different path in control flow (instead of trapping) when a pattern match failure happens". (You decide, but think of the target audience.)